akka-typed/src/main/scala/akka/typed/cluster/ClusterSharding.scala

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (C) 2009-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.cluster
+
+import akka.cluster.sharding.ShardCoordinator.ShardAllocationStrategy
+import akka.cluster.sharding.ClusterShardingSettings
+import akka.typed.{ ActorRef, ActorSystem, Behavior, Extension, ExtensionId, Props }
+
+sealed case class ShardingEnvelope[A](entityId: String, message: A)
+object StartEntity {
+  def apply[A](entityId: String): ShardingEnvelope[A] =
+    new ShardingEnvelope[A](entityId, null.asInstanceOf[A])
+}
+
+object TypedMessageExtractor {
```

Comment:
we should probably not qualify the classes with `Typed`, already a separate package

Reply:
Good point